sources/packages/backend/apps/api/src/services/disbursement-schedule/_tests_/unit/disbursement-schedule.getMaxTuitionRemittance.spec.ts

```diff
@@ -0,0 +1,104 @@
+import {
+  NotificationActionsService,
+  SequenceControlService,
+} from "@sims/services";
+import { DisbursementScheduleService } from "../../disbursement-schedule-service";
+import { DataSource } from "typeorm";
+import { addDays } from "@sims/utilities";
+import { COE_WINDOW } from "../../../../utilities";
+import { COEApprovalPeriodStatus } from "../../disbursement-schedule.models";
+
+describe("DisbursementScheduleService-getCOEApprovalPeriodStatus", () => {
+  let service: DisbursementScheduleService;
+
+  beforeAll(async () => {
+    const dataSource = {} as DataSource;
+    dataSource.getRepository = jest.fn();
+    const sequenceService: SequenceControlService =
+      {} as SequenceControlService;
+    const notificationActionsService: NotificationActionsService =
+      {} as NotificationActionsService;
+    service = new DisbursementScheduleService(
+      dataSource,
+      sequenceService,
+      notificationActionsService,
+    );
+  });
+
+  it("Should throw error when disbursementDate is undefined", () => {
+    // Arrange
+    const disbursementDate = undefined;
+    const studyEndDate = new Date();
+
+    // Act and Assert.
+    expect(() => {
+      service.getCOEApprovalPeriodStatus(disbursementDate, studyEndDate);
+    }).toThrow(
+      "disbursementDate and studyEndDate are required for COE window verification.",
+    );
+  });
+
+  it("Should throw error when both disbursementDate and studyEndDate is undefined", () => {
+    // Arrange
+    const disbursementDate = undefined;
+    const studyEndDate = undefined;
+
+    // Act and Assert.
+    expect(() => {
+      service.getCOEApprovalPeriodStatus(disbursementDate, studyEndDate);
+    }).toThrow(
+      "disbursementDate and studyEndDate are required for COE window verification.",
+    );
+  });
+
+  it(`Should return "${COEApprovalPeriodStatus.WithinApprovalPeriod}" when enrolment now within eligible approval period.`, () => {
```

Comment:
For the `WithinApprovalPeriod` we can have 2 tests:
- Should return "WithinApprovalPeriod" when enrolment now is the first day of the allowed COE window
- Should return "WithinApprovalPeriod" when enrolment now is the last day of the allowed COE window

Reply:
okay... the logic was checking for in-between, that's why I choose a random in-between date. but I get it. updating